packages/react-vapor/src/components/calendar/CalendarDay.tsx

```diff
@@ -30,50 +32,37 @@ export class CalendarDay extends React.Component<ICalendarDayProps> {
     }
 
     componentDidUpdate() {
-        if (
-            !this.props.day.isSelectable &&
-            this.props.day.isSelected &&
-            (this.props.day.isLowerLimit || this.props.day.isUpperLimit)
-        ) {
+        const {day} = this.props;
+
+        if (!day.isSelectable && day.isSelected && (day.isLowerLimit || day.isUpperLimit)) {
             this.props.onSelectUnselectable();
         }
     }
 
     render() {
-        const dayClasses: string[] = [];
-        const dayCellClasses: string[] = [CalendarDay.DEFAULT_DATE_CLASS];
-
-        if (!this.props.day.isCurrentMonth) {
-            dayClasses.push('other-month-date');
-        }
-
-        if (!this.props.day.isSelectable) {
-            dayCellClasses.push('un-selectable');
-        }
-
-        if (this.props.day.isToday) {
-            dayClasses.push('todays-date');
-        }
+        const {day} = this.props;
 
-        if (this.props.day.isSelected && this.props.day.isSelectable) {
-            dayClasses.push('selected-date');
+        const isSelectedAndSelectable = day.isSelected && day.isSelectable;
 
-            if (this.props.day.isLowerLimit) {
-                dayClasses.push('lower-limit');
-            }
+        const dayClassNames: string = classNames({
+            'other-month-date': !day.isCurrentMonth,
+            'todays-date': day.isToday,
+            countdown: day.isCountdown,
+            'selected-date': isSelectedAndSelectable,
```

Comment:
nitpick but :see_no_evil: 
```suggestion
            'selected-date': isSelectableAndSelectabled,
```

Reply:
:confused: tbh I don't undersand this suggestion.

Reply:
Selectedabled isn't a word, so I think my change is correct here

Reply:
oops I meant `isSelectableAndSelected` :see_no_evil: 

it's just switching the conditions because it needs to be selectable to be selected xD

Reply:
ha yeah that makes sense. On it